src/node/Launcher.ts

```diff
@@ -105,7 +108,16 @@ class ChromeLauncher implements ProductLauncher {
     }
 
     let chromeExecutable = executablePath;
-    if (!executablePath) {
+
+    if (channel) {
+      // executablePath is detected by channel, so it should not be specified by user.
+      assert(
+        !executablePath,
```

Comment:
```suggestion
        !(executablePath ||
      process.env.PUPPETEER_EXECUTABLE_PATH ||
      process.env.npm_config_puppeteer_executable_path ||
      process.env.npm_package_config_puppeteer_executable_path),
```

Reply:
Oh, thank you for your suggestion. I understood this assertion is just for checking invalid usage of the parameters (both executablePath and channel are specified), and I thought each environment-based variables are just ignored when channel is specified.

Also I think warning is enough to notify users to know PUPPETEER_EXECUTABLE_PATH or other variables are ignored.
I don't think it is invalid to specify channel when PUPPETEER_EXECUTABLE_PATH is set as a default executable path.
How do you think?

Reply:
Yes, while I think it would be super helpful to make sure that the parameters can not be used in an invalid and ambiguous way I feel like the same should also apply to the environment variables. 
Otherwise, people might be setting them and then wonder why they are just silently ignored...
Do you see any downsides to including them here?

Reply:
This is just my concern but I worry about the use-case to share the code with `channel: chrome` in team members.
Script developer assumes Chrome is launched by setting `channel: chrome`, but if some members set `export PUPPETEER_EXECUTABLE_PATH=/path/to/chrome` in .bashrc or another, they claims "script is not working!!" by the assertion error. 

> I feel like the same should also apply to the environment variables.
> Otherwise, people might be setting them and then wonder why they are just silently ignored...

I agree this behavior is really simple, however it is a little confusing in terms of consistency.
PUPPETEER_EXECUTABLE_PATH is already ignored when `executablePath` is explicitly specified, and some users would regard the environment variable as a definition of the *default value* of executablePath.


I know I worry a lot, so if simple behavior is preferred, I will add the assertion for environment variables.

Reply:
You have a good point about `PUPPETEER_EXECUTABLE_PATH` being used as a fall back.
Let's keep the behaviour the way you suggested!